base/special/trig.jl

```diff
@@ -860,8 +860,10 @@ function cospi(x::T) where T<:Union{Integer,Rational}
     end
 end
 
-sinpi(x::Integer) = x >= 0 ? zero(float(x)) : -zero(float(x))
-cospi(x::Integer) = isodd(x) ? -one(float(x)) : one(float(x))
+sinpi(x::T) where {T<:Integer} = zero(signed(T))
+sinpi(x::Bool) = 0
+cospi(x::T) where {T<:Integer} = isodd(x) ? -one(signed(T)) : one(signed(T))
+cospi(x::Bool) = x ? -1 : 1
```

Comment:
Why do you return `Int` (i.e. `Int32` or `Int64`)?
Should this be `Int8`?

Reply:
`signed(true)` returns `Int`, so I think that makes more sense here. With #35854, we shouldn't even need these.

Reply:
This traces back to [`Core`](https://github.com/JuliaLang/julia/blob/master/base/boot.jl#L737-L738). Then I guess this would be the least-invasive solution.

Reply:
Do we even need these specializations in the first place? It is `Bool <: Integer`, so I think we're good without them.

Reply:
We need them until #35854 is merged

Reply:
Done